app/grandchallenge/cases/image_builders/tiff.py

```diff
@@ -13,157 +15,129 @@
+        if not self.color_space:
+            raise ValidationError("ColorSpace not valid")
+        if not self.voxel_width_mm:
+            raise ValidationError("Voxel width could not be determined")
+        if not self.voxel_height_mm:
+            raise ValidationError("Voxel height could not be determined")
+
+
+def _get_tag_value(tags, tag):
+    try:
+        return tags[tag].value
+    except KeyError:
+        return None
 
 
-def load_tiff_file(*, path: Path) -> GrandChallengeTiffFile:
+def _get_voxel_spacing_mm(tags, tag):
     """
-    Loads and validates a file using tifffile
-    :param path: The path to the potential tiff file
-    :return: A tiff file that can be used in the rest of grand challenge
-    """
-    try:
-        file = tifffile.TiffFile(str(path.absolute()))
-    except ValueError:
-        raise ValidationError("Image isn't a TIFF file")
+    Calculate the voxel spacing in mm.
 
-    tags = _validate_tifffile(pages=file.pages)
+    Use the set of tags from the tiff image to calculate the spacing for a
+    particular dimension. Supports INCH and CENTIMETER resolution units.
 
-    return GrandChallengeTiffFile(path=path, tags=tags)
+    Parameters
+    ----------
+    tags
+        The collection of tags from the tif file.
+    tag
+        The tag that contains the resolution of the tif file along the
+        dimension of interest.
 
+    Raises
+    ------
+    ValidationError
+        Raised if an unrecognised resolution unit is used.
 
-def _validate_tifffile(  # noqa: C901
-    *, pages: tifffile.tifffile.TiffPages
-) -> GrandChallengeTiffFileTags:
-    """
-    Validates a tiff file loaded with tifffile for use in grand challenge
-    :param pages: The pages and tags from tiffile
-    :return: The extracted tags that are needed by the rest of the framework
-    """
 
-    def get_tag_value(tags, tag, required=True):
-        try:
-            return tags[tag].value
-        except KeyError:
-            if required:
-                raise ValidationError(
-                    f"Tiff file is missing required tag {tag}"
-                )
-
-    def get_voxel_spacing_mm(tags, tag):
-        """
-        Calculate the voxel spacing in mm.
-
-        Use the set of tags from the tiff image to calculate the spacing for a
-        particular dimension. Supports INCH and CENTIMETER resolution units.
+    Returns
+    -------
+        The voxel spacing in mm.
 
-        Parameters
-        ----------
-        tags
-            The collection of tags from the tif file.
-        tag
-            The tag that contains the resolution of the tif file along the
-            dimension of interest.
+    """
+    try:
+        # Resolution is a tuple of the number of pixels and the length of the
+        # image in cm or inches, depending on resolution unit
+        resolution_unit = str(_get_tag_value(tags, "ResolutionUnit"))
+        resolution = _get_tag_value(tags, tag)
```

Comment:
This can return `None` so calling `resolution[0] / resolution[1]` below could also raise a `TypeError`. If the list containes less than 2 items it could also raise an `IndexError`. Should we also add an except for those?